lib/methods/findAll.js

```diff
@@ -4,5 +4,13 @@
  * @param {Function} callback Callback passed an Error object (or null if successful) and the models.
  */
 exports.findAll = function (Model, callback) {
-	Model._query({limit: 1000}, callback, Model);
+	var options = {
+		limit: 1000
+	};
+
+	if(Model.metadata && typeof Model.metadata.response_json_depth !== 'undefined') {
```

Comment:
According to the ArrowDB docs `response_json_depth` should be a number, so please explicitly check for that instead of just if it's not undefined.